service/test/y2storage/agama_proposal_search_test.rb

```diff
@@ -0,0 +1,221 @@
+        vda3_sid = probed.find_by_name("/dev/vda3").sid
+
+        devicegraph = proposal.propose
+
+        system = devicegraph.find_by_name("/dev/system")
+        expect(system.lvm_pvs.map(&:blk_device).map(&:sid)).to contain_exactly(vda2_sid, vda3_sid)
+      end
+    end
+
+    context "when marking several partitions for resizing" do
+      let(:config_json) do
+        {
+          boot:   { configure: false },
+          drives: [
+            {
+              search:     disk_name,
+              partitions: [
+                { search: search, size: { min: 0, max: "current" } },
+                { size: "25 GiB", filesystem: { path: "/" } }
+              ]
+            }
+          ]
+        }
+      end
+
+      before do
+        allow_any_instance_of(Y2Storage::Partition)
+          .to(receive(:detect_resize_info))
+          .and_return(resize_info)
+      end
+
+      let(:resize_info) do
+        instance_double(
+          Y2Storage::ResizeInfo, resize_ok?: true,
+          min_size: Y2Storage::DiskSize::GiB(4), max_size: Y2Storage::DiskSize::GiB(35)
+        )
+      end
+
+      shared_examples "resize" do
+        it "resizes several partitions if needed" do
+          probed = Y2Storage::StorageManager.instance.probed
+          vda2_size = probed.find_by_name("/dev/vda2").size
+          vda3_size = probed.find_by_name("/dev/vda3").size
+
+          devicegraph = proposal.propose
+
+          expect(devicegraph.find_by_name("/dev/vda2").size).to be < vda2_size
+          expect(devicegraph.find_by_name("/dev/vda3").size).to be < vda3_size
+        end
+      end
+
+      context "using an empty search to match the partitions" do
+        let(:search) { {} }
+
+        context "if there are several partitions at the disk" do
+          let(:disk_name) { "/dev/vda" }
+
+          include_examples "resize"
+        end
+
+        context "if there are no partitions in the disk" do
+          let(:disk_name) { "/dev/vdc" }
+
+          it "register an error and returns nil" do
+            expect(proposal.propose).to be_nil
+            expect(proposal.issues_list).to include an_object_having_attributes(
+              description: /mandatory partition/,
+              severity:    Agama::Issue::Severity::ERROR
+            )
+          end
+        end
+      end
+
+      context "using asterisk as the search to match the partitions" do
+        let(:search) { "*" }
+
+        context "if there are several partitions at the disk" do
+          let(:disk_name) { "/dev/vda" }
+
+          include_examples "resize"
+        end
+
+        context "if there are no partitions in the disk" do
+          let(:disk_name) { "/dev/vdc" }
+
+          it "processes the proposal" do
+            devicegraph = proposal.propose
+            disk = devicegraph.find_by_name(disk_name)
+            expect(disk.partitions.size).to eq 1
+          end
+
+          it "register a warning about non-existent partitions" do
```

Comment:
BTW, I think we should reconsider this behavior. I don't think it's worth a warning. It can generate noise in many situations.

Reply:
For the record: we agreed it makes no sense and it will be changed in a subsequent pull request.